db-migration/run.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+echo 'Logging into Gov PaaS'
+
+if cf login -a api.london.cloud.service.gov.uk -u ${CF_USERNAME} -p ${CF_PASSWORD} -s ${CF_SPACE}; then
+  echo 'Logged into Gov PaaS successfully'
+else
+  echo 'ERROR could not log into Gov Paas'
+  exit 1
+fi
+
+echo "Setting up conduit into Gov PaaS DB and making dump file"
+
+if cf conduit ${CF_SERVICE} -- pg_dump --file dumpfile --no-acl --encoding utf8 --clean --no-owner --if-exists -Fc; then
+  echo 'Dump file of PaaS DB created successfully'
+else
+  echo 'ERROR trying to conduit into the PaaS DB or creating the dump file'
+  exit 1
+fi
+
+while [ ! -e  "$dumpfile"]; do
+  echo "Waiting for dump file to be ready for pg restore"
```

Comment:
I noticed on my local machine that sometimes the dump file is still increasing in size on the drive after the command has finished, so I'm hoping this will allow the file to be fully created before proceeding to use it for the restore

Reply:
Doesn't the `-e` expression just check for file existence? Surely that won't help with checking its size has stabilised?

Reply:
It does! My thinking was that this would be sufficient that by the time we've seen it exists it will probably be ready pretty quickly after that for use. It's a non-trivial problem to work out when the file is finished being written to however 😢, one option I know of is `lsof` (but not tried this), keeping checking the size of the file, or just sleep for an arbitrary amount of time to give it breathing space 😁. We can try one of these or just see how we get on, not seen any issues with the above yet but it's more likely to occur when the DBs are big!

Reply:
Fair enough, worth discussing with Rachael I think if you haven't already.

Reply:
Ok I think I've made an improvement to this, by first checking the file exists (thought it might be useful to keep this anyway) and then running a `fuser` command on a loop to check if the dumpfile is being accessed by any processes. Testing hasn't shown if this definitely useful as the dumpfile is so quick to write it doesn't end up reporting anything, but we can see how we get on when we do a staging migration